### PR TITLE
Switch from go dep to go mods for dependencies

### PR DESCRIPTION
**Pull request by fedepaol**
# Description

This pr removes the vendor folder switches to go modules for handling dependencies.

Fixes # (issue)

## Type of change

Please select the appropiate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Built and ran the test

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [x] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

**Comment by fedepaol** [comment]
Here I only have the bare 'go.mod' / 'go.sum' files.
@cynepco3hahue / @davidvossel do you think it's worh having a local vendor folder created by go mod vendor?

**Comment by cynepco3hahue** [comment]
@fedepaol good question, I believe it can be useful when you want to save some time on compilation once you clone the project or if you want to use the code from the place where you do not have access to the internet.
At least for KubeVirt we used it because we needed to build it d/s.

File: features/sctp/test/go.mod [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	go4.org v0.0.0-20191010144846-132d2879e1e9 // indirect
+	golang.org/x/oauth2 v0.0.0-20191202225959-858c2ad4c8b6 // indirect
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
+	k8s.io/api v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v11.0.0+incompatible
+	k8s.io/utils v0.0.0-20191114200735-6ca3b61696b6 // indirect
+)
+
+replace github.com/openshift/machine-config-operator => github.com/openshift/machine-config-operator v0.0.1-0.20191204220237-5e6c4cabf194
```
Comment (cynepco3hahue) [inline features/sctp/test/go.mod]: you can join all `replace` directives via
```go
replace (
    ....
)
```

Comment (fedepaol) [reply features/sctp/test/go.mod]: Will do, thanks!

Comment (fedepaol) [reply features/sctp/test/go.mod]: Done

**Comment by fedepaol** [comment]
@cynepco3hahue should be "/lgtm" I think

**Comment by cynepco3hahue** [comment]
@fedepaol We still do not have permissions to review or merge code, we should create OWNERS file and add our names under it.

**Comment by fedepaol** [comment]
@cynepco3hahue ah! you are right. Will open another pr for that :-)

**Comment by fedepaol** [comment]
> /lgtm
> 
> Can we get a CI test soon that's at least performing a build here to ensure the build doesn't inadvertently break

@davidvossel we'll have as soon as https://github.com/openshift/release/pull/6147 gets in

**Comment by fedepaol** [comment]
Looks like it's time to put an `OWNERS` file here.

**Comment by fedepaol** [comment]
Oh, flakyness
/retest

**Comment by davidvossel** [comment]
```
sctp_test.go:11:2: cannot find package "github.com/onsi/gomega" in any of:
	/usr/local/go/src/github.com/onsi/gomega (from $GOROOT)
	/go/src/github.com/onsi/gomega (from $GOPATH)
FAIL	github.com/openshift-kni/baremetal-deploy/features/sctp/test [setup failed]
```

missing a dependency

**Comment by fedepaol** [comment]
> ```
> sctp_test.go:11:2: cannot find package "github.com/onsi/gomega" in any of:
> 	/usr/local/go/src/github.com/onsi/gomega (from $GOROOT)
> 	/go/src/github.com/onsi/gomega (from $GOPATH)
> FAIL	github.com/openshift-kni/baremetal-deploy/features/sctp/test [setup failed]
> ```
> 
> missing a dependency

I _think_ (let's see the result of the run) that I was missing go get before.
Since I am not go mod vendor-ing here, a go get is required to fetch the required modules.

**Comment by fedepaol** [comment]
Ok, we need to use the local `vendor` folder:

```
sh-4.2$ go get
go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
        ignoring go.mod;
        see 'go help modules'
```

**Comment by fedepaol** [comment]
We could also have this fixed by setting `GO111MODULE=on`, but the more I think about this, the more I am convinced we should use vendor to stick to what we have for executables.

**Comment by cynepco3hahue** [comment]
> Ok, we need to use the local `vendor` folder:
> 
> ```
> sh-4.2$ go get
> go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
>         ignoring go.mod;
>         see 'go help modules'
> ```

I think you can use  `GO111MODULE=on go get`

**Comment by fedepaol** [comment]
> > Ok, we need to use the local `vendor` folder:
> > ```
> > sh-4.2$ go get
> > go get: warning: modules disabled by GO111MODULE=auto in GOPATH/src;
> >         ignoring go.mod;
> >         see 'go help modules'
> > ```
> 
> I think you can use `GO111MODULE=on go get`

Yes, see my note https://github.com/openshift-kni/baremetal-deploy/pull/40#issuecomment-563133061

File: features/sctp/Makefile [hunk 2]
```diff
@@ -1,7 +1,13 @@
 .PHONY: test, ginkgo
 
+deps:
+	cd test && \
+	go get && \
```
Comment (cynepco3hahue) [inline features/sctp/Makefile]: better to use `go mod tidy` it will fetch needed dependencies and clean unused one

**Pull request merged**